source/solvers/gls_sharp_navier_stokes.cc

```diff
@@ -1951,6 +1943,253 @@ GLSSharpNavierStokesSolver<dim>::copy_local_rhs_to_global_rhs(
+                "omega_x", particles[i_particle].omega[0]);
+              particles_information_table.set_precision("omega_x", 12);
+              particles_information_table.add_value(
+                "omega_y", particles[i_particle].omega[1]);
+              particles_information_table.set_precision("omega_y", 12);
+            }
+          particles_information_table.add_value("omega_z",
+                                                particles[i_particle].omega[2]);
+          particles_information_table.set_precision("omega_z", 12);
+          if (dim == 3)
+            {
+              particles_information_table.add_value(
+                "T_x", particles[i_particle].torques[0]);
+              particles_information_table.set_precision("T_x", 12);
+              particles_information_table.add_value(
+                "T_y", particles[i_particle].torques[1]);
+              particles_information_table.set_precision("T_y", 12);
+            }
+          particles_information_table.add_value(
+            "T_z", particles[i_particle].torques[2]);
+          particles_information_table.set_precision("T_z", 12);
+        }
+      // Write the table in the checkpoint file.
+      particles_information_table.write_text(output);
+    }
+}
+
+template <int dim>
+void
+GLSSharpNavierStokesSolver<dim>::read_checkpoint()
+{
+  this->GLSNavierStokesSolver<dim>::read_checkpoint();
+
+  TimerOutput::Scope t(this->computing_timer,
+                       "Reset Nitsche solid mesh and particles");
+
+  std::string prefix =
+    this->simulation_parameters.simulation_control.output_folder +
+    this->simulation_parameters.restart_parameters.filename;
+
+  std::string filename =
+    this->simulation_parameters.simulation_control.output_folder + prefix +
+    ".ib_particles";
+
+  // refill the table from checkpoint
+  for (unsigned int p_i = 0; p_i < particles.size(); ++p_i)
+    {
+      std::string filename_table =
+        this->simulation_parameters.simulation_control.output_folder +
+        this->simulation_parameters.particlesParameters.ib_force_output_file +
+        "." + Utilities::int_to_string(p_i, 2) + ".dat";
+      fill_table_from_file(table_p[p_i], filename_table);
+    }
+
+  // Read the data of each particle and put the relevant information in a
+  // vector.
+  std::pair<std::vector<std::string>, std::vector<std::vector<double>>>
+    restart_data;
+  fill_vectors_from_file(restart_data, filename);
+
+  std::vector<double> p_x =
+    restart_data.second[std::find(restart_data.first.begin(),
+                                  restart_data.first.end(),
+                                  "p_x") -
+                        restart_data.first.begin()];
+  std::vector<double> p_y =
+    restart_data.second[std::find(restart_data.first.begin(),
+                                  restart_data.first.end(),
+                                  "p_y") -
+                        restart_data.first.begin()];
```

Comment:
I don't really like this copy paste of stuff. Maybe a good strategy could have been to have a map where they key is the string and the value is a reference to the vector. This way the std::find could be inside of a loop over the map.
In all cases, this is something we can try later to clean once this is merged in a follow-up PR.